--- wormux-0.8alpha1/configure.old	2006-10-26 12:50:32.000000000 +0200
+++ wormux-0.8alpha1/configure	2006-10-26 12:51:07.000000000 +0200
@@ -6811,9 +6811,7 @@
             GCC_FLAGS="$GCC_FLAGS -Wno-unused-parameter"
 
     if test "x${debug}" = "xyes"; then
-        GCC_FLAGS="$GCC_FLAGS -O0 -g -DDEBUG"
-    else
-        GCC_FLAGS="$GCC_FLAGS -O2"
+        GCC_FLAGS="$GCC_FLAGS -DDEBUG"
     fi
 
             CFLAGS="$GCC_FLAGS"
