--- configure	
+++ configure	
@@ -51684,7 +51684,8 @@
 fi
 
 
-    gnash_boost_version=""
+    gnash_
+boost_version=""
   gnash_boost_topdir=""
   gnash_boost_libdir=""
   missing_headers=""
@@ -51782,9 +51783,9 @@
         missing_libs=""
       fi
       for j in ${boost_libs}; do
-        dirs="`ls -dr $i/libboost_${j}*.${shlibext} $i/libboost_${j}*.a 2>/dev/null`"
+        dirs="`ls -dr $i/libboost_${j}*.${shlibext} $i/libboost_${j}*.so 2>/dev/null`"
         if test -n "${dirs}"; then
-          libname="`echo ${dirs} | sed -e 's:^.*/lib::' -e "s:\.${shlibext}::" -e "s:\.a::"`"
+          libname="`echo ${dirs} | sed -e 's:^.*/lib::' -e "s:\.${shlibext}::" -e "s:\.so::"`"
           if test x$dirname = x; then
             dirname="`echo ${dirs} | sed -e 's:/libboost.*$::'`"
            if test x"${dirname}" != "x/usr/lib"; then
@@ -51798,9 +51799,9 @@
       done
     done
     for j in ${extra_boost_libs}; do
-      dirs="`ls -dr ${dirname}/libboost_${j}*.${shlibext} ${dirname}/libboost_${j}*.a 2>/dev/null`"
+      dirs="`ls -dr ${dirname}/libboost_${j}*.${shlibext} ${dirname}/libboost_${j}*.so 2>/dev/null`"
       if test -n "${dirs}"; then
-          libname="`echo ${dirs} | sed -e 's:^.*/lib::' -e "s:\.${shlibext}::" -e "s:\.a::"`"
+          libname="`echo ${dirs} | sed -e 's:^.*/lib::' -e "s:\.${shlibext}::" -e "s:\.so::"`"
         ac_cv_path_boost_extra_lib="${ac_cv_path_boost_extra_lib} -l${libname}"
       else
         extra_missing_libs="${extra_missing_libs} $j"
