diff -Naur Celestia-e13024d723463388f8d35ca16e2b3b8a6f906789.orig/src/celutil/formatnum.cpp Celestia-e13024d723463388f8d35ca16e2b3b8a6f906789/src/celutil/formatnum.cpp
--- Celestia-e13024d723463388f8d35ca16e2b3b8a6f906789.orig/src/celutil/formatnum.cpp	2024-12-18 11:53:05.000000000 +0100
+++ Celestia-e13024d723463388f8d35ca16e2b3b8a6f906789/src/celutil/formatnum.cpp	2024-12-31 12:29:19.018426549 +0100
@@ -58,7 +58,7 @@
         if (it != ctx.end() && *it != '}')
         {
 #if FMT_VERSION >= 100100
-            throw_format_error("invalid format");
+            report_error("invalid format");
 #else
             assert(0);
             return ctx.end();
diff -Naur Celestia-e13024d723463388f8d35ca16e2b3b8a6f906789.orig/src/celutil/formatnum.h Celestia-e13024d723463388f8d35ca16e2b3b8a6f906789/src/celutil/formatnum.h
--- Celestia-e13024d723463388f8d35ca16e2b3b8a6f906789.orig/src/celutil/formatnum.h	2024-12-18 11:53:05.000000000 +0100
+++ Celestia-e13024d723463388f8d35ca16e2b3b8a6f906789/src/celutil/formatnum.h	2024-12-31 12:29:23.325070063 +0100
@@ -122,7 +122,7 @@
         if (it != ctx.end() && *it != '}')
         {
 #if FMT_VERSION >= 100100
-            throw_format_error("invalid format");
+            report_error("invalid format");
 #else
             assert(0);
             return ctx.end();
