--- mediastreamer2/src/nowebcam.c	2008/03/13 22:44:22	1.12
+++ mediastreamer2/src/nowebcam.c	2008/04/29 13:52:48	1.13
@@ -70,7 +70,9 @@
 	sws_ctx=sws_getContext(av_context.width,av_context.height,PIX_FMT_YUV420P,
 		reqsize->width,reqsize->height,PIX_FMT_YUV420P,SWS_FAST_BILINEAR,
                 NULL, NULL, NULL);
-	sws_scale(sws_ctx,orig.data,orig.linesize,0,0,dest.data,dest.linesize);
+	if (sws_scale(sws_ctx,orig.data,orig.linesize,0,av_context.height,dest.data,dest.linesize)<0){
+		ms_error("jpeg2yuv: sws_scale() failed.");
+	}
 	sws_freeContext(sws_ctx);
 	avcodec_close(&av_context);
 	return ret;
--- mediastreamer2/src/pixconv.c	2008/03/13 22:44:22	1.12
+++ mediastreamer2/src/pixconv.c	2008/04/29 13:52:48	1.13
@@ -139,7 +139,7 @@
                 		NULL, NULL, NULL);
 			}
 			if (sws_scale(s->sws_ctx,inbuf.data,inbuf.linesize, 0,
-				0, s->outbuf.planes, s->outbuf.strides)!=0){
+				s->size.height, s->outbuf.planes, s->outbuf.strides)<0){
 				ms_error("MSPixConv: Error in sws_scale().");
 			}
 			freemsg(im);
--- mediastreamer2/src/sizeconv.c	2008/03/13 22:44:21	1.13
+++ mediastreamer2/src/sizeconv.c	2008/04/29 13:52:48	1.14
@@ -160,7 +160,7 @@
 				struct SwsContext *sws_ctx=get_resampler(s,inbuf.w,inbuf.h);
 				mblk_t *om=size_conv_alloc_mblk(s);
 				if (sws_scale(sws_ctx,inbuf.planes,inbuf.strides, 0,
-					0, s->outbuf.planes, s->outbuf.strides)!=0){
+					inbuf.h, s->outbuf.planes, s->outbuf.strides)<0){
 					ms_error("MSSizeConv: error in sws_scale().");
 				}
 				ms_queue_put(f->outputs[0],om);
--- mediastreamer2/src/videodec.c	2008/03/14 20:59:09	1.22
+++ mediastreamer2/src/videodec.c	2008/04/29 13:52:48	1.23
@@ -241,7 +241,7 @@
                 	NULL, NULL, NULL);
 	}
 	if (sws_scale(s->sws_ctx,orig->data,orig->linesize, 0,
-					0, s->outbuf.planes, s->outbuf.strides)!=0){
+					ctx->height, s->outbuf.planes, s->outbuf.strides)<0){
 		ms_error("%s: error in sws_scale().",f->desc->name);
 	}
 	return dupmsg(s->yuv_msg);
--- mediastreamer2/src/videoout.c	2008/03/13 22:44:21	1.28
+++ mediastreamer2/src/videoout.c	2008/04/29 13:52:48	1.29
@@ -355,7 +355,7 @@
 			src->w,src->h,PIX_FMT_RGB24,
 			0, NULL, NULL, NULL);
 	if (sws_scale(sws,src->planes,src->strides, 0,
-           			0, &p, &rgb_stride)!=0){
+           			0, &p, &rgb_stride)<0){
 			ms_error("Error in 420->rgb sws_scale().");
 	}
 	sws_freeContext(sws);
@@ -665,7 +665,7 @@
 					       obj->local_pic.w,obj->local_pic.h);
 		}
 		if (sws_scale(obj->sws2,src.planes,src.strides, 0,
-			      0, obj->local_pic.planes, obj->local_pic.strides)!=0){
+			      src.h, obj->local_pic.planes, obj->local_pic.strides)<0){
 		  ms_error("Error in sws_scale().");
 		}
 	      }
@@ -683,7 +683,7 @@
 			}
 			ms_display_lock(obj->display);
 			if (sws_scale(obj->sws1,src.planes,src.strides, 0,
-            			0, obj->fbuf.planes, obj->fbuf.strides)!=0){
+            			src.h, obj->fbuf.planes, obj->fbuf.strides)<0){
 				ms_error("Error in sws_scale().");
 			}
 			ms_display_unlock(obj->display);
