--- tools/v2/tools/qt4.jam
+++ tools/v2/tools/qt4.jam
@@ -61,8 +61,8 @@
     {
         .initialized = true ;
         .prefix = $(prefix) ;
-	.includepath = $(prefix)/include ;
-	.librarypath = $(prefix)/lib ;
+	.includepath = $(prefix)/include/qt4 ;
+	.librarypath = $(prefix)/lib/qt4 ;
 
         # Generates cpp files from header files using "moc" tool
         generators.register-standard qt4.moc : H : CPP(moc_%) ;
