diff -Naur pyvcloud-12.orig/pyvcloud/vapp.py pyvcloud-12/pyvcloud/vapp.py
--- pyvcloud-12.orig/pyvcloud/vapp.py	2015-04-21 14:30:01.028814009 +0200
+++ pyvcloud-12/pyvcloud/vapp.py	2015-04-21 14:30:30.623826056 +0200
@@ -21,7 +21,6 @@
 from schema.vcd.v1_5.schemas.vcloud.taskType import TaskType
 from schema.vcd.v1_5.schemas.vcloud.vAppType import VAppType, NetworkConnectionSectionType
 from iptools import ipv4, IpRange
-from tabulate import tabulate
 from pyvcloud.helper import CommonUtils
 
 
diff -Naur pyvcloud-12.orig/requirements.txt pyvcloud-12/requirements.txt
--- pyvcloud-12.orig/requirements.txt	2015-04-21 14:30:57.183836813 +0200
+++ pyvcloud-12/requirements.txt	2015-04-21 14:30:23.550823183 +0200
@@ -1,8 +1,4 @@
-PyYAML==3.10
-iptools==0.6.1
-requests==2.4.3
-tabulate==0.7.3
-wsgiref==0.1.2
-xmltodict==0.9.0
-lxml==3.4.1
-netaddr==0.7.13
+iptools>=0.6.1
+lxml>=3.4.1
+netaddr>=0.7.13
+requests>=2.4.3
