diff -Naur allegro-4.2.0/src/gp2x/gp2x_joy.c allegro-4.2.0_new/src/gp2x/gp2x_joy.c
--- allegro-4.2.0/src/gp2x/gp2x_joy.c	2006-08-11 19:09:18.000000000 +0200
+++ allegro-4.2.0_new/src/gp2x/gp2x_joy.c	2006-08-11 19:22:21.000000000 +0200
@@ -10,14 +10,18 @@
 };
 
 static struct gp2x_joystick_button gp2x_joystick_buttons[] = {
-	{	'C',	13,		"B"		},
-	{	'C',	14,		"X"		},
-	{	'C',	15,		"Y"		},
-	{	'C',	12,		"A"		},
-	{	'C',	11,		"R"		},
-	{	'C',	10,		"L"		},
-	{	'D',	11,		"Click"	},
-	{	0,		0,		NULL	}
+	{	'C',	13,		"B"			},
+	{	'C',	14,		"X"			},
+	{	'C',	15,		"Y"			},
+	{	'C',	12,		"A"			},
+	{	'C',	11,		"R"			},
+	{	'C',	10,		"L"			},
+	{   'C',    9,      "Select"	},
+	{   'C',    8,      "Start"		},
+	{   'D',    6,      "Vol Down"	},
+	{   'D',    7,      "Vol Up"	},
+	{	'D',	11,		"Click"		},
+	{	0,		0,		NULL		}
 };
 
 
