--- fgrun-1.6.2/CMakeLists.txt.orig	2013-02-19 13:16:55.556582089 +0100
+++ fgrun-1.6.2/CMakeLists.txt	2013-02-19 13:18:14.882574299 +0100
@@ -87,7 +87,7 @@
 find_package(ZLIB REQUIRED)
 find_package(FLTK REQUIRED)
 find_package(OpenSceneGraph 2.8.1 REQUIRED osgText osgDB osgParticle osgUtil osgViewer osgGA)
-find_package(SimGear 2.8.0 REQUIRED)
+find_package(SimGear 2.10.0 REQUIRED)
 if (ENABLE_NLS)
     find_package(Gettext)
 endif (ENABLE_NLS)
