diff -Naur DVDStyler-1.6.1.orig/configure.in DVDStyler-1.6.1/configure.in
--- DVDStyler-1.6.1.orig/configure.in	2008-06-22 12:00:51.000000000 +0200
+++ DVDStyler-1.6.1/configure.in	2008-06-22 12:01:14.000000000 +0200
@@ -84,7 +84,7 @@
 AC_CHECK_LIB(avformat, av_register_all, [], [AC_MSG_ERROR([*** missing ffmpeg library: libavformat])])
 AC_CHECK_LIB(avcodec, avcodec_decode_audio2, [], [AC_MSG_ERROR([*** missing ffmpeg library: libavcodec])])
 AC_CHECK_LIB(swscale, sws_getContext)
-AC_CHECK_HEADERS(ffmpeg/avcodec.h, [], [AC_MSG_ERROR([*** missing ffmpeg header file: ffmpeg/avcodec.h.])])
+AC_CHECK_HEADERS(libavcodec/avcodec.h, [], [AC_MSG_ERROR([*** missing ffmpeg header file: libavcodec/avcodec.h.])])
 
 CXXFLAGS="$CXXFLAGS -DDATADIR=\\\"$datadir/$PACKAGE\\\""
 
diff -Naur DVDStyler-1.6.1.orig/src/mediaenc_ffmpeg.cpp DVDStyler-1.6.1/src/mediaenc_ffmpeg.cpp
--- DVDStyler-1.6.1.orig/src/mediaenc_ffmpeg.cpp	2008-06-22 12:00:51.000000000 +0200
+++ DVDStyler-1.6.1/src/mediaenc_ffmpeg.cpp	2008-06-22 12:02:19.000000000 +0200
@@ -24,8 +24,8 @@
 extern "C" {
 #define __STDC_CONSTANT_MACROS
 #define __STDC_LIMIT_MACROS
-#include <ffmpeg/avformat.h>
-#include <ffmpeg/swscale.h>
+#include <libavformat/avformat.h>
+#include <libswscale/swscale.h>
 }
 
 #define AUDIO_BUF_SIZE 50000
