--- Makefile.orig	2008-07-18 11:11:20.000000000 +0200
+++ Makefile	2008-07-18 11:11:51.000000000 +0200
@@ -9,12 +9,6 @@
 GOBJECT_CFLAGS=`pkg-config --cflags gobject-2.0`
 GOBJECT_LIBS=`pkg-config --libs gobject-2.0`
 
-ifdef DEBUG
-CFLAGS+=-ggdb
-else
-CFLAGS+=-O2
-endif
-
 EXTRA_WARNINGS=-Wall -Wextra -Wformat-nonliteral -Wcast-align -Wpointer-arith \
 	       -Wbad-function-cast -Wmissing-prototypes -Wstrict-prototypes \
 	       -Wmissing-declarations -Winline -Wundef -Wnested-externs -Wcast-qual \
@@ -32,7 +26,7 @@
 # For glib < 2.6 support (libpurple maniacs)
 FALLBACK_CFLAGS+=-I./fix_purple
 
-LDFLAGS:=-Wl,--no-undefined
+LDFLAGS+=-Wl,--no-undefined
 
 purpledir=$(DESTDIR)/$(PURPLE_PREFIX)/lib/purple-2
 
