http://sourceforge.net/p/hugin/hugin/ci/aae44c4b0570e2cfc609e84d6e74d38a31453b73/

diff -Naur hugin-2013.0.0.orig/doc/deghosting_mask.pod hugin-2013.0.0/doc/deghosting_mask.pod
--- hugin-2013.0.0.orig/doc/deghosting_mask.pod	2013-09-09 01:46:06.127737748 +0200
+++ hugin-2013.0.0/doc/deghosting_mask.pod	2013-09-09 01:46:28.737736742 +0200
@@ -1,3 +1,5 @@
+=encoding utf8
+
 =head1 NAME
 
 deghosting_mask - Creates mask for removing ghosting in images
