diff -Naur wine-4.4.orig/tools/winebuild/main.c wine-4.4/tools/winebuild/main.c
--- wine-4.4.orig/tools/winebuild/main.c	2019-03-15 21:18:31.000000000 +0100
+++ wine-4.4/tools/winebuild/main.c	2019-03-17 12:16:04.199209848 +0100
@@ -47,10 +47,13 @@
 int force_pointer_size = 0;
 int unwind_tables = 0;
 
+#undef FORCE_POINTER_SIZE
 #ifdef __i386__
 enum target_cpu target_cpu = CPU_x86;
+#define FORCE_POINTER_SIZE
 #elif defined(__x86_64__)
 enum target_cpu target_cpu = CPU_x86_64;
+#define FORCE_POINTER_SIZE
 #elif defined(__powerpc__)
 enum target_cpu target_cpu = CPU_POWERPC;
 #elif defined(__arm__)
@@ -619,6 +622,10 @@
     signal( SIGTERM, exit_on_signal );
     signal( SIGINT, exit_on_signal );
 
+#ifdef FORCE_POINTER_SIZE
+    force_pointer_size = sizeof(size_t);
+#endif
+
     argv = parse_options( argc, argv, spec );
     atexit( cleanup );  /* make sure we remove the output file on exit */
 
diff -Naur wine-4.4.orig/tools/winegcc/winegcc.c wine-4.4/tools/winegcc/winegcc.c
--- wine-4.4.orig/tools/winegcc/winegcc.c	2019-03-15 21:18:31.000000000 +0100
+++ wine-4.4/tools/winegcc/winegcc.c	2019-03-17 12:15:20.264214885 +0100
@@ -222,10 +222,13 @@
     strarray* files;
 };
 
+#undef FORCE_POINTER_SIZE
 #ifdef __i386__
 static const enum target_cpu build_cpu = CPU_x86;
+#define FORCE_POINTER_SIZE
 #elif defined(__x86_64__)
 static const enum target_cpu build_cpu = CPU_x86_64;
+#define FORCE_POINTER_SIZE
 #elif defined(__powerpc__)
 static const enum target_cpu build_cpu = CPU_POWERPC;
 #elif defined(__arm__)
@@ -1324,6 +1327,9 @@
     opts.linker_args = strarray_alloc();
     opts.compiler_args = strarray_alloc();
     opts.winebuild_args = strarray_alloc();
+#ifdef FORCE_POINTER_SIZE
+    opts.force_pointer_size = sizeof(size_t);
+#endif
 
     /* determine the processor type */
     if (strendswith(argv[0], "winecpp")) opts.processor = proc_cpp;
