--- 3dfsb-1.2/cmake/Modules/FindGStreamer.cmake.orig	2015-02-09 04:20:00.000000000 +0100
+++ 3dfsb-1.2/cmake/Modules/FindGStreamer.cmake	2019-12-05 10:48:17.640478581 +0100
@@ -56,7 +56,7 @@
 macro(FIND_GSTREAMER_COMPONENT _component_prefix _pkgconfig_name _library)
 
     string(REGEX MATCH "(.*)>=(.*)" _dummy "${_pkgconfig_name}")
-    if ("${CMAKE_MATCH_2}" STREQUAL "")
+    if ("${CMAKE_MATCH_2}" STREQUAL "" AND (NOT ${GStreamer_FIND_VERSION} STREQUAL ""))
         pkg_check_modules(PC_${_component_prefix} "${_pkgconfig_name} >= ${GStreamer_FIND_VERSION}")
     else ()
         pkg_check_modules(PC_${_component_prefix} ${_pkgconfig_name})
