--- 3dfsb-1.2/CMakeLists.txt.orig	2019-12-05 11:01:08.603789775 +0100
+++ 3dfsb-1.2/CMakeLists.txt	2019-12-05 11:01:14.789792047 +0100
@@ -31,14 +31,14 @@
 
 # Generate xpm images from png images
 IF(NOT MOGRIFY)
-   EXEC_PROGRAM(mogrify ARGS -format xpm images/*.png
+   EXEC_PROGRAM(mogrify ${CMAKE_SOURCE_DIR} ARGS -format xpm images/*.png
                 OUTPUT_VARIABLE MOGRIFY)
    SET(MOGRIFY "${MOGRIFY}" CACHE STRING "Mogrify")
 ENDIF(NOT MOGRIFY)
 
 # Generate .h files from resource files
 IF(NOT XXD)
-   EXEC_PROGRAM(xxd ARGS -i resources/xorg.conf resources/xorg.conf.h
+   EXEC_PROGRAM(xxd ${CMAKE_SOURCE_DIR} ARGS -i resources/xorg.conf resources/xorg.conf.h
                 OUTPUT_VARIABLE XXD)
    SET(XXD "${XXD}" CACHE STRING "XXD")
 ENDIF(NOT XXD)
