diff -Nur dangerdeep-0.4.0+git.1449654226.360eb39/src/model.cpp new/src/model.cpp
--- dangerdeep-0.4.0+git.1449654226.360eb39/src/model.cpp	2015-12-09 10:43:46.000000000 +0100
+++ new/src/model.cpp	2017-03-26 14:31:19.956437814 +0200
@@ -47,6 +47,7 @@
 /* bounding sphere for a triangle: diameter is longest edge, chose center accordingly */
 
 texture::mapping_mode model::mapping = texture::LINEAR_MIPMAP_LINEAR;//texture::NEAREST;
+const unsigned model::mesh::no_adjacency;
 
 unsigned model::init_count = 0;
 
