--- SConstruct
+++ SConstruct
@@ -124,6 +124,7 @@
 
 else:
 	print "Compiling for Unix/Posix/Linux Environment"
+	SConsignFile()
 	env = Environment(ENV = os.environ)
 	env.Append(CPPPATH = ['/usr/include/SDL', '/usr/include/GL'])
 	libpath = ['/usr/X11R6/lib']
@@ -177,6 +178,11 @@
 		else:
 			print 'Using x86 SSE/MMX optimizations.'
 	env.Append(CCFLAGS = ccflags)
+	env.Replace(CXX = os.environ['CXX'])
+	if os.environ.get('CXXFLAGS'):
+		env.Replace(CCFLAGS = os.environ['CXXFLAGS'])
+	if os.environ.get('LDFLAGS'):
+		env.Replace(LINKFLAGS = os.environ['LDFLAGS'])
 	datadir = '/usr/local/share/dangerdeep'
 	build_dir = 'linux'
 	# check for broken libGL, ignore undefined symbols then
