--- dangerdeep-9999.orig/SConstruct	2014-10-30 20:10:41.864713576 +0100
+++ dangerdeep-9999/SConstruct	2014-10-30 20:10:37.231709639 +0100
@@ -208,9 +208,13 @@
 	# Set libpath for real
 	libpath = [archlib, archlib + '/X11', '/usr/X11R6/lib']
 
-	ccflags += ' -march=' + arch
-
 	env.Append(LINKFLAGS = '`pkg-config --libs-only-L x11`')
+	if os.environ.has_key('CXX'):
+		env['CXX'] = os.environ['CXX']
+	if os.environ.has_key('CXXFLAGS'):
+		ccflags += os.environ['CXXFLAGS']
+	if os.environ.has_key('LDFLAGS'):
+		env.Append(LINKFLAGS = os.environ['LDFLAGS'])
 	if (debug >= 4):
 		ccflags += ' -g -pg -O3' # profiling
 	elif (debug == 3):
@@ -221,7 +225,6 @@
 		env.Append(CPPDEFINES = ['DEBUG'])
 	elif (debug == 1):
 		# same as debug==0 but with DEBUG defined to make logging work (and probably other stuff)
-		ccflags += ' -g -O2'
 		env.Append(CPPDEFINES = ['DEBUG'])
 	elif (debug == -1):
 		ccflags += ' -g -O3 -mfpmath=sse -mmmx -msse -m3dnow'
@@ -235,8 +238,6 @@
 			env.Replace(CXX = "icpc")
 			ccflags += ' -ip -w '
 			build_dir = 'linux-icc'
-		else:
-			ccflags += ' -g -O2' # debug symbols will be stripped by the linker for a debian package
 
 	if (useefence >= 1):
 		extralibs += ['efence']
