diff -Naur src.orig/mymain.cpp src/mymain.cpp
--- src.orig/mymain.cpp	2009-11-17 14:28:45.000000000 +0100
+++ src/mymain.cpp	2009-11-17 14:28:57.000000000 +0100
@@ -65,8 +65,6 @@
 		result = -2;
 	}
 	log::instance().write(std::cerr, log::LOG_SYSINFO);
-	std::ofstream f("log.txt");
-	log::instance().write(f, log::LOG_SYSINFO);
 	return result;
 }
 
diff -Naur src.orig/system.cpp src/system.cpp
--- src.orig/system.cpp	2009-11-17 14:28:45.000000000 +0100
+++ src/system.cpp	2009-11-17 14:28:51.000000000 +0100
@@ -458,10 +458,6 @@
 				case SDL_QUIT:			// Quit event
 					log_info("---------- immediate exit ----------");
 					log::instance().write(std::cerr, log::LOG_SYSINFO);
- 					{
- 						std::ofstream f("log.txt");
- 						log::instance().write(f, log::LOG_SYSINFO);
-					}
 					throw quit_exception(0);
 				
 				case SDL_ACTIVEEVENT:		// Application activation or focus event
